### ISSUE TITLE
got error Bucket Creation Failed: serde_xml: ParseIntError: invalid digit found in string  with AWS IAM s3

### ISSUE THREAD

**Issue by uderik** [IBM/core-dump-handler#126]
Hi! I'am install [core-dump-handler](https://github.com/IBM/core-dump-handler) with AWS IAM S3 support, but on starts got error Bucket Creation Failed: serde_xml: ParseIntError: invalid digit found in string 

what i do wrong?

```
[2023-01-12T11:55:18Z INFO  core_dump_agent] Starting sysctl for fs.suid_dumpable /var/mnt/core-dump-handler/suid_dumpable.bak with 2                                               
[2023-01-12T11:55:18Z INFO  core_dump_agent] Getting sysctl for fs.suid_dumpable                                                                                                    
[2023-01-12T11:55:18Z INFO  core_dump_agent] Created Backup of /var/mnt/core-dump-handler/suid_dumpable.bak                                                                         
fs.suid_dumpable = 2                                                                                                                                                                
[2023-01-12T11:55:18Z INFO  core_dump_agent] Creating /var/mnt/core-dump-handler/.env file with LOG_LEVEL=Debug                                                                     
[2023-01-12T11:55:18Z INFO  core_dump_agent] Writing composer .env                                                                                                                  
    LOG_LEVEL=Debug                                                                                                                                                                 
    IGNORE_CRIO=false                                                                                                                                                               
    CRIO_IMAGE_CMD=img                                                                                                                                                              
    USE_CRIO_CONF=false                                                                                                                                                             
    FILENAME_TEMPLATE={uuid}-dump-{timestamp}-{hostname}-{exe_name}-{pid}-{signal}                                                                                                  
    LOG_LENGTH=500                                                                                                                                                                  
    POD_SELECTOR_LABEL=                                                                                                                                                             
    TIMEOUT=600                                                                                                                                                                     
    COMPRESSION=true                                                                                                                                                                
    CORE_EVENTS=false                                                                                                                                                               
    EVENT_DIRECTORY=/var/mnt/core-dump-handler/events                                                                                                                               
                                                                                                                                                                                    
[2023-01-12T11:55:18Z INFO  core_dump_agent] Executing Agent with location : /var/mnt/core-dump-handler/cores                                                                       
[2023-01-12T11:55:19Z ERROR core_dump_agent] Bucket Creation Failed: serde_xml: ParseIntError: invalid digit found in string                                                        
[2023-01-12T11:55:19Z INFO  core_dump_agent] INotify Starting...                                                                                                                    
[2023-01-12T11:55:19Z INFO  core_dump_agent] INotify Initialised...                                                                                                                 
[2023-01-12T11:55:19Z INFO  core_dump_agent] INotify watching : /var/mnt/core-dump-handler/cores     
``` 


and with debug:

```
[2023-01-12T11:59:20Z DEBUG serde_xml_rs::de] Peeked StartElement({https://sts.amazonaws.com/doc/2011-06-15/}Expiration, {"": "https://sts.amazonaws.com/doc/2011-06-15/", "xml": "h
ttp://www.w3.org/XML/1998/namespace", "xmlns": "http://www.w3.org/2000/xmlns/"})                                                                                                    
[2023-01-12T11:59:20Z DEBUG serde_xml_rs::de] Peeked StartElement({https://sts.amazonaws.com/doc/2011-06-15/}Expiration, {"": "https://sts.amazonaws.com/doc/2011-06-15/", "xml": "h
ttp://www.w3.org/XML/1998/namespace", "xmlns": "http://www.w3.org/2000/xmlns/"})                                                                                                    
[2023-01-12T11:59:20Z DEBUG serde_xml_rs::de] Peeked StartElement({https://sts.amazonaws.com/doc/2011-06-15/}Expiration, {"": "https://sts.amazonaws.com/doc/2011-06-15/", "xml": "h
ttp://www.w3.org/XML/1998/namespace", "xmlns": "http://www.w3.org/2000/xmlns/"})                                                                                                    
[2023-01-12T11:59:20Z DEBUG serde_xml_rs::de] Peeked StartElement({https://sts.amazonaws.com/doc/2011-06-15/}Expiration, {"": "https://sts.amazonaws.com/doc/2011-06-15/", "xml": "h
ttp://www.w3.org/XML/1998/namespace", "xmlns": "http://www.w3.org/2000/xmlns/"})                                                                                                    
[2023-01-12T11:59:20Z DEBUG serde_xml_rs::de] Peeked StartElement({https://sts.amazonaws.com/doc/2011-06-15/}Expiration, {"": "https://sts.amazonaws.com/doc/2011-06-15/", "xml": "h
ttp://www.w3.org/XML/1998/namespace", "xmlns": "http://www.w3.org/2000/xmlns/"})                                                                                                    
[2023-01-12T11:59:20Z DEBUG serde_xml_rs::de] Fetched StartElement({https://sts.amazonaws.com/doc/2011-06-15/}Expiration, {"": "https://sts.amazonaws.com/doc/2011-06-15/", "xml": "
http://www.w3.org/XML/1998/namespace", "xmlns": "http://www.w3.org/2000/xmlns/"})                                                                                                   
[2023-01-12T11:59:20Z DEBUG serde_xml_rs::de] Peeked Characters(2023-01-12T12:59:20Z)                                                                                               
[2023-01-12T11:59:20Z DEBUG serde_xml_rs::de] Fetched Characters(2023-01-12T12:59:20Z)                                                                                              
[2023-01-12T11:59:20Z DEBUG serde_xml_rs::de] Fetched EndElement({https://sts.amazonaws.com/doc/2011-06-15/}Expiration)                                                             
[2023-01-12T11:59:20Z ERROR core_dump_agent] Bucket Creation Failed: serde_xml: ParseIntError: invalid digit found in string  
```

**Comment by uderik** [comment]
downgrade to quay.io/icdh/core-dump-handler:v8.7.0 version fix my issue

**Comment by No9** [comment]
Thanks @uderik 
This looks like a serialization problem with the [rust-s3](https://github.com/durch/rust-s3) as we bumped to 0.32.3 from 0.31 version between 8.7 and 8.9 ans we didn't really change the app logic around our bucket creation and our integration tests have been passing successfully. 

Although our integration tests don't test against AWS.

I'll raise it over in that repo and see if we can get any additional thoughts. 
In the mean time I will provide a downgraded release for you to test and validate it is the the version bump.

**Comment by No9** [comment]
@uderik 
Once this build is finished https://quay.io/repository/icdh/core-dump-handler/build/fb2962c1-abf8-4add-a12b-b16b935d52a3 
If you have time can you checkout this branch https://github.com/IBM/core-dump-handler/tree/agent-s3-down and validate if it works or not. The image tag has been updated to point to the build so it should drop in directly to your current workflow.

**Comment by No9** [comment]
Thanks @uderik 
Just to confirm I understand you correctly - it's the `agent-s3-down` branch that works for you? 
Not the `v8.9.0` release that's just started working.

**Comment by uderik** [comment]
@No9 yes it's works

**Comment by No9** [comment]
Great - I'll spin a 8.9.1 release out of that this weekend 
And confirm the issue back over in the downstream project.

**Comment by No9** [comment]
Released in v8.10.0 so I'm closing this. 
Will watch https://github.com/durch/rust-s3/issues/322 to see when an upgrade is possible.
